src/core/infer_request.cc

```diff
@@ -820,6 +860,34 @@ InferenceRequest::Input::DataBuffer(
   return Status::Success;
 }
 
+Status
+InferenceRequest::Input::DataBufferForHostPolicy(
+    const size_t idx, const void** base, size_t* byte_size,
+    TRITONSERVER_MemoryType* memory_type, int64_t* memory_type_id,
+    const std::string host_policy_name) const
+{
+  auto device_data = host_policy_data_map_.find(host_policy_name);
+  if (device_data == host_policy_data_map_.end()) {
+    // Return data buffer if there is no host-policy specific buffer available
+    *base = data_->BufferAt(idx, byte_size, memory_type, memory_type_id);
+  } else {
+    *base = device_data->second->BufferAt(
+        idx, byte_size, memory_type, memory_type_id);
+  }
+
+  return Status::Success;
+}
+
+size_t
+InferenceRequest::Input::DataBufferCountForHostPolicy(
+    const std::string host_policy_name) const
+{
+  auto policy_data = host_policy_data_map_.find(host_policy_name);
+  if (policy_data != host_policy_data_map_.end()) {
+    return policy_data->second->BufferCount();
+  }
+  return 0;
```

Comment:
We need to have (and clearly document) standard handling for the "host policy" APIs for the case when the host policy is not explicitly initialized for the request. It seems like we want to always fall back to non-host-policy data? If so then we need to do so consistently and we need to document that having a non-host-policy copy of the data is always required.

Reply:
Agree, I also think that falling back to non-host-policy data in new functions will simplify the handling [here](https://github.com/triton-inference-server/server/pull/2898/files#diff-a62653eb1ec25ffe4a68c842b42039f1e4749c46c732f230dd2fa32e5ec10e72R962-R967) by always calling to the new function.